pilot/pkg/serviceregistry/kube/controller/controller.go

```diff
@@ -1317,6 +1317,11 @@ func (c *Controller) getProxyServiceInstancesByPod(pod *v1.Pod,
 	var out []*model.ServiceInstance
 
 	for _, svc := range c.servicesForNamespacedName(kube.NamespacedNameForK8sObject(service)) {
+		// Check if service is visible. Service can be invisible if exportTo is set to '~' for example.
+		// We should consider service instances that are only visible to pod.
+		if proxy.LastPushContext != nil && !proxy.LastPushContext.IsServiceVisible(svc, pod.Namespace) {
```

Comment:
unnecessary, the pod itself is the service instance

Reply:
see my description of the PR. without this, we get inbound listener conflict if an there are multiple service select same pods. Any way to prevent that?

Reply:
I do not think it is a reasonable case

Reply:
Say you define two services for a proxy, and you do not want to use one.

Reply:
But we want to explicitly say we do not want to use it by specifying "networking.istio.io": "~" on the service that is not meant to be used for this purpose.

Thinking aloud, may be we we should also merge service instances by port to prevent this inbound conflicts?

Reply:
Changed it to not complain if service instance ports are equal